### PR TITLE
Fix test that's breaking pre-release workflow

### PR DESCRIPTION
**Pull request by nemanjaglumac**
We had to adjust E2E tests when we removed H2 support. That was done in #33212.

But this broke the pre-release workflow.
That workflow doesn't have necessary docker services needed for this test (postgres and mysql).

On top of that, adding both databases in this test is kind of redundant, and definitely an overkill. We just want to make sure we can add a database. So, why not use SQLite that doesn't require external services?

This PR accomplishes exactly that.
It simplifies and speeds up this particular test by using SQLite database.

### Before
Example of a broken pre-release from my fork:
image

image

### After

Screenshot from a local run
image

### CODE REVIEW

File: e2e/test/scenarios/onboarding/setup/setup.cy.spec.js [hunk 1]
```diff
@@ -234,69 +234,64 @@ describe("scenarios > setup", () => {
     });
   });
 
-  ["postgres", "mysql"].forEach(engine => {
-    it(`should allow you to connect a ${engine} db during setup`, () => {
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:238]: @iethree please note that even if we decided to stick with these two databases, we have to always include the `@external` tag in the test config to signal that it depends on external services.

Comment (iethree) [reply e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:238]: sorry, my bad on forgetting the @external tag. I think we really wanted coverage on connecting our main transactional DBs during setup, can we do them in addition to sqlite?

Comment (nemanjaglumac) [reply e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:238]: We already have E2E test that's covering that.
https://github.com/metabase/metabase/blob/master/e2e/test/scenarios/admin/databases/add-new-database.cy.spec.js

Comment (nemanjaglumac) [reply e2e/test/scenarios/onboarding/setup/setup.cy.spec.js:238]: > sorry, my bad on forgetting the @external tag.

No biggie.
Btw, that's not what broke the pre-release. It's the fact that it doesn't have QA DBs, nor `QA_DB_ENABLED` set to true.

**Review by iethree — APPROVED** [review]

**Pull request merged**